### PR TITLE
add wasm files that had gone missing

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

The files where the wasm is compiled from via rust had gone missing. I have added them back in.

When adding them back in, I was hit by:

```git add wasmfromrust/src/
The following paths are ignored by one of your .gitignore files:
pytket/tests/wasm-generation/wasmfromrust/src
hint: Use -f if you really want to add them.
hint: Turn this message off by running
hint: "git config advice.addIgnoredFile false"
```

But I have not found any ignores that would explain this. Any ideas?

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by cqc-alec** [comment]
Seems these files were deleted in https://github.com/CQCL/tket/pull/1070/files when perhaps they should have been moved.

Maybe they are in a global gitignore or in `.git/info/exclude` on your system?

**Review by cqc-alec — APPROVED** [review]

File: pytket/tests/wasm-generation/wasmfromrust/src/lib-2.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
+// src/lib.rs
```
Comment (cqc-melf) [inline pytket/tests/wasm-generation/wasmfromrust/src/lib-2.rs:1]:
```suggestion
#![deny(warnings)]
// src/lib.rs

File: pytket/tests/wasm-generation/wasmfromrust/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// src/lib.rs
```
Comment (cqc-melf) [inline pytket/tests/wasm-generation/wasmfromrust/src/lib.rs:1]:
```suggestion
#![deny(warnings)]
// src/lib.rs

**Comment by cqc-melf** [comment]
I have updated this PR do add the rust version of "-Werror"

**Pull request merged**